Describe new order types (#670)
vegaprotocol/documentation: docs/concepts/trading-on-vega/orders.md

* If the reference price no longer exists (e.g. no best bid)
* If the price moves to a value that means it would create an invalid order if the offset was applied

### Iceberg order
An iceberg order is a limit order for a large amount that, rather than being entered as a single large order of that size, is placed on the book as a smaller order that is replenished as that order amount is filled. The peak / 'visible' amount can be filled with one trade, while the reserve is used to support the smaller order amount.

> In other words, an iceberg order allows for the smaller order amount, known as the peak size, to be replenished from a 'hidden' total size. An iceberg's peak size is replenished in full until the whole iceberg's volume trades away, or the order is cancelled or expires. It's replenished whenever the peak drops below the minimum visible size, set when submitting the order. If the reserve amount becomes lower than the peak size, the peak size is then set to the full remaining amount, with nothing in reserve.

Reviewer: If we're saying the "peak_size" is the static value which is the most the "visible size" can be, should here:
"It's replenished whenever the peak drops below the minimum visible size"
it instead be:
"It's replenished whenever the remaining visible size drops below the minimum visible size"?

Reviewer: Maybe also here:
"An iceberg's peak size is replenished in full"

should instead be:
"An iceberg's visible size is replenished in full to the peak size"

When the iceberg order enters the order book passively, it trades like a typical persistent order, where the peak size is the displayed size.

On refresh, the replenished order will get a new time priority, as it's treated like a new order.

#### Amend an iceberg order
An iceberg order can be amended. The remaining amount can be increased or decreased, while the peak size will stay the same. If the amended remaining amount is smaller than the peak size, the peak size is automatically reduced to be the same as the amended remaining amount.

Amending an iceberg does not affect the peak size's time priority.

### Stop order
A stop order is an order to buy or sell once the price reaches a specified price, known as the trigger price. Stop orders can be used to help a trader limit losses (stop loss), or capitalise on a gain (take profit) automatically when they already have an open position.

Stop orders can only be used to bring a trader's position closer to zero, and thus make use of the [reduce only](#reduce-only) condition and will only be accepted if the trader already has an open position on the market.

> A stop order must include a trigger price - for a take profit stop order, or percentage - for a trailing stop, and whether to deploy the order once the last traded price "rises above" or "falls below" the given stop price/percentage. It can, but doesn't need to, have an expiry date and time, or an execution time (???) at which point the order will be filled if possible.

Reviewer 2: trigger price can be for take profit or stop loss (probably easiest to just say along the lines of "stop order must include a trigger price, for a fixed level, or percentage, for a trailing stop,"

Author: Thanks, I've changed that. And what does an "execution time" mean? I included that but wasn't sure how that differed from an expiry.

> A stop order must include a trigger price - for a take profit stop order, or percentage - for a trailing stop, and whether to deploy the order once the last traded price "rises above" or "falls below" the given stop price/percentage. It can, but doesn't need to, have an expiry date and time, or an execution time (???) at which point the order will be filled if possible.

Author: ```suggestion
A stop order must include a trigger price at a fixed level, or it can be a percentage for a trailing stop. It must also include whether to deploy the order once the last traded price "rises above" or "falls below" the given stop price/percentage. It can, but doesn't need to, have an expiry date and time, or an execution time at which point the order will be filled, if possible.
```

While a stop order can be a single instruction, it could alternatively be a pair of stop orders where one cancels the other (OCO). Using the OCO option allows a trader to have a stop loss and a take profit/trailing stop instruction for the same position. If one of the pair is triggered, cancelled, deleted, or rejected, the other one is automatically cancelled.

> If a trader's position size moves to zero and there are no open orders, the trader's stop orders for that position are cancelled.

Reviewer 2: ```suggestion
If a trader's position size moves to zero and there are no open orders, the trader's stop orders for that market are cancelled.
```

> There's a limit to how many stop orders that one public key can have active at one time, set by the 'spam.protection.max.stopOrdersPerMarket' network parameter.

Reviewer 2: ```suggestion
There's a limit to how many stop orders any one public key can have active at one time, set by the 'spam.protection.max.stopOrdersPerMarket' network parameter.
```
